docs/source/api-ref/session.rst

```diff
@@ -92,64 +92,83 @@ can be used for that user.
 
 .. code-block:: python
 
-   # Generation of a certificate and private key using the subprocess module to
-   # make direct calls to openssl at the shell level.  This assumes that
-   # openssl is installed on the system.
-
-   from subprocess import Popen, CalledProcessError, PIPE
-   from cobra.mit.access import MoDirectory
-   from cobra.mit.session import LoginSession
-   from cobra.mit.request import ConfigRequest
-   from cobra.model.pol import Uni as PolUni
-   from cobra.model.aaa import UserEp as AaaUserEp
-   from cobra.model.aaa import User as AaaUser
-   from cobra.model.aaa import UserDomain as AaaUserDomain
-   from cobra.model.aaa import UserRole as AaaUserRole
-   from cobra.model.aaa import UserCert as AaaUserCert
-
-   certUser = 'myuser'
-   pKeyFile = 'myuser.key'
-   certFile = 'myuser.cert'
-
-   # Generate the certificate in the current directory
-   cmd = ["openssl", "req", "-new", "-newkey", "rsa:1024", "-days", "36500",
-          "-nodes", "-x509", "-keyout", pKeyFile, "-out", certFile,
-          "-subj", "/CN=Generic/O=Acme/C=US"]
-   proc = Popen(cmd, stdin=PIPE, stdout=PIPE, stderr=PIPE)
-   out, error = proc.communicate()
-   # If an error occurs, fail
-   if proc.returncode != 0:
-       print("Output: {0}, Error {1}".format(out, error))
-       raise CalledProcessError(proc.returncode, " ".join(cmd))
-
-   # At this point pKeyFile and certFile exist as files in the local directory.
-   # pKeyFile will be used when we want to generate signatures.  certFile is
-   # contains the X.509 certificate (with public key) that needs to be pushed
-   # to the APIC for a local user.
-
-   with open(certFile, "r") as file:
-       PEMdata = file.read()
-
-   # Generate a local user to commit to the APIC
-   polUni = PolUni('')
-   aaaUserEp = AaaUserEp(polUni)
-   aaaUser = AaaUser(aaaUserEp, certUser)
-   aaaUserDomain = AaaUserDomain(aaaUser, name='all')
-   # Other aaaUserRoles maybe needed to give the user other privileges
-   aaaUserRole = AaaUserRole(aaaUserDomain, name='read-all',
-                             privType='readPriv')
-   # Attach the certificate to that user.
-   aaaUserCert = AaaUserCert(aaaUser, certUser + '-cert')
-   # Using the data read in from the certificate file.
-   aaaUserCert.data = PEMdata
-
-   # Push the new local user to the APIC
-   session = LoginSession('https://10.1.1.1', 'admin', 'ins3965!', secure=False)
-   moDir = MoDirectory(session)
-   moDir.login()
-   cr = ConfigRequest()
-   cr.addMo(aaaUser)
-   moDir.commit(cr)
+    >>> # Generation of a certificate and private key using the subprocess module to
+    ... # make direct calls to openssl at the shell level.  This assumes that
+    ... # openssl is installed on the system.
+    >>>
```

Comment:
There needs to be a ... after every comment because that's what the python interpreter does:

    >>> # Generation of a certificate and private key using the subprocess module to
    ... # make direct calls to openssl at the shell level.  This assumes that
    ... # openssl is installed on the system.
    ... 
    >>>